controllers/chat/group/leaveGroupMember.js

```diff
@@ -15,26 +15,29 @@ const leaveGroupMember = async (req, res) => {
   const ownUser = await UsersFunction.findUserById(User, userId);
 
   try {
+    let channelResponse;
+
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const currentChannel = await client.channel(CHANNEL_TYPE_STRING.GROUP, channelId);
+    const queriedChannel = await client.queryChannels({
+      type: CHANNEL_TYPE_STRING.GROUP,
+      id: channelId
+    });
+    console.log(':::queriedChannel', queriedChannel);
+    let all_members = queriedChannel[0].data.better_channel_member.map((member) => member.user.id);
+
     const response = await Getstream.chat.removeGroupMember(channelId, userId);
     const {channel, channelApiResponse} = response.data || {};
 
-    let channelResponse;
-    try {
-      const {newChannelName, betterChannelMember, betterChannelMemberObject, updatedChannel} =
-        await BetterSocialCore.chat.updateBetterChannelMembers(channel, channelApiResponse, true);
-
-      const client = new StreamChat(
-        Environment.GETSTREAM_API_KEY,
-        Environment.GETSTREAM_API_SECRET
-      );
-      const currentChannel = await client.channel(CHANNEL_TYPE_STRING.GROUP, channelId);
+    const {newChannelName, betterChannelMember, betterChannelMemberObject, updatedChannel} =
+      await BetterSocialCore.chat.updateBetterChannelMembers(channel, channelApiResponse, true);
 
-      const textOwnUser = `You left this group`;
-      const textTargetUser = `${ownUser.username} left this group`;
-      const textDefaultUser = `${ownUser.username} left this group`;
-      const members = betterChannelMember.map((member) => member.user_id);
+    const textOwnUser = `You left this group`;
+    const textTargetUser = `${ownUser.username} left this group`;
+    const textDefaultUser = `${ownUser.username} left this group`;
 
-      await currentChannel.sendMessage({
+    await currentChannel.sendMessage(
+      {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a `queryChannels` call to get the channel details. However, it seems like this might be unnecessary as you already have a `currentChannel` object from the previous line which should contain all the necessary information about the channel. If the `better_channel_member` data is not available in the `currentChannel` object, then this operation is justified. Otherwise, it's an unnecessary API call that could impact performance.

```diff
-    const queriedChannel = await client.queryChannels({
-      type: CHANNEL_TYPE_STRING.GROUP,
-      id: channelId
-    });
-    console.log(':::queriedChannel', queriedChannel);
-    let all_members = queriedChannel[0].data.better_channel_member.map((member) => member.user.id);
+    let all_members = currentChannel.data.better_channel_member.map((member) => member.user.id);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->